code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -439,20 +435,33 @@ This is here to make the tiles around the station mininuke change when it's arme
 	poi_list |= src
 	START_PROCESSING(SSobj, src)
 
-/obj/item/weapon/disk/nuclear/process()
-	var/turf/disk_loc = get_turf(src)
-	if(!disk_loc || disk_loc.z > ZLEVEL_CENTCOM)
-		get(src, /mob) << "<span class='danger'>You can't help but feel that you just lost something back there...</span>"
-		qdel(src)
+/obj/item/weapon/disk/nuclear/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is \
+		going delta! It looks like they're comitting suicide.</span>")
+	playsound(user.loc, 'sound/machines/Alarm.ogg', 50, -1, 1)
+	var/end_time = world.time + 100
+	var/orig_color = user.color
+	while(world.time < end_time)
+		if(!user)
+			return
+		user.color = RANDOM_COLOUR
+		sleep(1)
+	user.color = orig_color
+	user.visible_message("<span class='suicide'>[user] was destroyed \
+		by the nuclear blast!</span>")
+	return OXYLOSS
```

Comment:
Change it to TOXLOSS|BRUTELOSS|FIRELOSS to simulate being exploded/burnt/irradiated